core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -1238,13 +1241,138 @@ components:
           $ref: "#/components/schemas/TransactionHeader"
         instructions:
           type: string
-          description: The decompiled transaction manifest instructions. Only returned if enabled in TransactionFormatOptions on your request.
+          description: The decompiled transaction manifest instructions. Only returned if enabled in `TransactionFormatOptions` on your request.
         blobs_hex:
           type: object
-          description: A map of the hex-encoded blob hash, to hex-encoded blob content. Only returned if enabled in TransactionFormatOptions on your request.
+          description: A map of the hex-encoded blob hash, to hex-encoded blob content. Only returned if enabled in `TransactionFormatOptions` on your request.
           additionalProperties:
             type: string
             description: The hex-encoded blob content
+        message:
+          description: The optional transaction message. Only returned if present and enabled in `TransactionFormatOptions` on your request.
+          $ref: "#/components/schemas/TransactionMessage"
+    TransactionMessageType:
+      type: string
+      enum:
+        - Plaintext
+        - Encrypted
+    TransactionMessage:
+      type: object
+      required:
+        - type
+      properties:
+        type:
+          $ref: "#/components/schemas/TransactionMessageType"
+      discriminator:
+        propertyName: type
+        mapping:
+          # NOTE: These need to match TransactionMessageType
+          Plaintext: '#/components/schemas/PlaintextTransactionMessage'
+          Encrypted: '#/components/schemas/EncryptedTransactionMessage'
+    PlaintextTransactionMessage:
+      allOf:
+        - $ref: "#/components/schemas/TransactionMessage"
+        - type: object
+          description: An unencrypted message.
+          required:
+            - mime_type
+            - content
+          properties:
+            mime_type:
+              type: string
+              description: The RFC 2046 MIME type of the `content`.
```

Comment:
It might be worth clarifying that this field is untrusted, eg:
 "Intended to represent the RFC 2046 MIME type of the `content`. A client cannot trust that this field is a valid mime type."